ext/wry_ruby/src/application/tray_id.rs

```diff
@@ -0,0 +1,50 @@
+use magnus::{function, Error, ExceptionClass, Module, Object, RString};
+use std::cell::RefCell;
+use wry::application::TrayId as TrayIdImpl;
+
+#[derive(Clone, Debug)]
+#[magnus::wrap(class = "TrayId")]
+pub struct TrayId {
+    inner: Option<RefCell<TrayIdImpl>>,
+}
+
+impl TrayId {
+    pub const EMPTY: TrayId = TrayId { inner: None };
+
+    pub fn new(unique_string: RString) -> Result<Self, Error> {
+        if unique_string.is_empty() {
+            return Ok(TrayId::EMPTY);
+        }
+
+        let s = match unique_string.as_interned_str() {
+            Some(str_ref) => str_ref.as_str().unwrap(),
+            None => {
+                println!("ok why is it none, hm check freeze or no");
+                return Err(Error::new(
+                    ExceptionClass::default(),
+                    "Interned string not available",
+                ));
+            }
+        };
```

Comment:
here we can use the unsafe method `as_str` on `RString`. It should be safe to do so. It's only unsafe if we yield back control to the ruby vm, but here we never do while this reference is alive, so we should be all good.

Reply:
oo okay